documents/ARCHITECTURE.md

```diff
@@ -0,0 +1,80 @@
+# Stickian Empire - The Architecture
+
+This document holds the key architectural components that the developers of the Stickian Empire game follow. The reading is dense, and the writer is denser, but don't worry, nothing goes over his head, his reflexes are too quick.
+
+Take with a hint of salt any point given as a draft with the status :page_facing_up:.
+
+Work in progress... :warning: :construction_worker:
+
+## Index
+
+1. [High-Level Architecture](#high-level-architecture)
+2. [Tools](#tools)
+3. [Back-end Architecture](#back-end-architecture)
+4. [The NoSQL Approach](#the-nosql-approach)
+5. [Real(bothersome)-time problem](#realbothersome-time-problem)
+6. ["Micro" Front-ends](#micro-front-ends)
+7. [Front-end state management](#front-end-state-management)
+
+## High-Level Architecture
+
+To be able to provide fun to thousands, if millions, of (long distant future) players, a robust and future proof architecture is necessary. This first High-Level approach is to guide the developers and contributors to our final goal.
+
+```mermaid
+flowchart LR
+    subgraph frontend
+        home[Login/Home screen]
+        game[Game screen]
+        game[Game screen]
+        game[Game screen]
+        chat[Chat view]
+    end
+    subgraph backend
+        home --- auths
+        auths[Authentication service] --- relationaldb
+        game --- gameb1
+        game --- gameb2
+        game --- gameb3
+        gameb1[Game service X] --- gamedb
+        gameb2[Game service Y] --- gamedb
+        gameb3[Game service Z] --- gamedb
+        chat --- chatb
+        chatb[Chat service] --- relationaldb
+        gamedb[(NoSQL Database)]
+        relationaldb[(Relational Database)]
+    end
+```
+
+## Tools
+
+The tools for this project are stated as follows:
+
+- [Go](https://go.dev/) as one of the simplest back-end language with an enourmous power, but a compiled environment to ensure types and no runtime hiccups (yes, I'm looking at you Python 3.x :eyes:).
```

Comment:
I agree with you that `Go` as tremendous power, specific for backend development. This seems a good option. However, thinking on the long term, any new joiner backend developer would have to know `Go`, which might become more difficult than finding a python developer. Nevertheless, we don't know the future trends of these programming languages yet, so we can take `Go` for an initial implementation.

Reply:
I take this argument as one in favor of `Go`. For a new joined, or a new developer for the matter of fact, `Go` is one of the easiest learning curves (I'd even dare say easier than python - since it doesn't have the crazy untyped pitfalls). Another advantage is, since `Go` holds a very good standard package and formatting guidelines, it's easier to conform with the rest of the code base. Last but not least, it's compilable and you can run it anywhere without any interpreter :eyes: